src/main/java/io/github/thebusybiscuit/mobcapturer/PelletListener.java

```diff
@@ -22,14 +26,22 @@ public PelletListener(MobCapturer plugin) {
     @EventHandler
     public void onProjectileHit(EntityDamageByEntityEvent e) {
         if (e.getDamager() instanceof Snowball && e.getEntity() instanceof LivingEntity && e.getDamager().hasMetadata("mob_capturing_cannon")) {
-            Optional<ItemStack> optional = plugin.capture((LivingEntity) e.getEntity());
+            Snowball s = (Snowball) e.getDamager();
 
-            if (optional.isPresent()) {
-                e.getDamager().removeMetadata("mob_capturing_cannon", plugin);
-                e.getEntity().remove();
-                e.getEntity().getWorld().dropItemNaturally(((LivingEntity) e.getEntity()).getEyeLocation(), optional.get());
+            if (canCapture((Player) s.getShooter(), e.getEntity().getLocation())) {
```

Comment:
What if Shooter is not an instance of Player...?
You are casting this unsafely here and never do any instanceof checks